### ISSUE TITLE
Missing in_proj_weight in attn_state

### ISSUE THREAD

**Issue by freshjh** [wyczzy/AIGI-Holmes#9]
In function _convert_to_huggingface()_, 

    
```
attn_state = self.model.model.model.visual.transformer.resblocks[i].attn.state_dict()
            
q = attn_state["in_proj_weight"][:1024, :].cpu()
k = attn_state["in_proj_weight"][1024:1024 * 2, :].cpu()
v = attn_state["in_proj_weight"][1024 * 2:1024 * 3, :].cpu()
o = attn_state['out_proj.weight'].cpu()
```

But **in_proj_weight** is missing in **attn_state**, 
```     
loratorch.register_model_param_after_backward(self.model)
```
should be added to networks/trainer/optimize_parameters()，after the gradient update (**only for lora training**, not for NPR training).

```
def optimize_parameters(self):
    self.forward()
    if self.opt.trainmode == "rine":
        self.loss = self.loss_fn(self.output[0].squeeze(1), self.label)
    else:
        self.loss = self.loss_fn(self.output.squeeze(1), self.label)
    self.optimizer.zero_grad()
    self.loss.backward()
    self.optimizer.step()
    loratorch.register_model_param_after_backward(self.model)
```

Ref: 
https://github.com/Baijiong-Lin/LoRA-Torch/issues/7
[Step 3 in quick start of LoRA-Torch](https://github.com/Baijiong-Lin/LoRA-Torch?tab=readme-ov-file#quick-start)
